core/sdk-core/src/main/java/software/amazon/awssdk/core/interceptor/Context.java

```diff
@@ -65,19 +69,35 @@ private Context() {}
     @ThreadSafe
     @SdkPublicApi
     public interface AfterMarshalling extends BeforeMarshalling {
+    }
+
+    /**
+     * The state of the execution when the {@link ExecutionInterceptor#modifyHttpRequest} method is invoked.
+     */
+    @ThreadSafe
+    @SdkPublicApi
+    public interface ModifyHttpRequest extends AfterMarshalling {
+    }
+
+    /**
+     * The state of the execution with the {@link ExecutionInterceptor#modifyHttpContent} method is invoked.
+     */
+    @ThreadSafe
+    @SdkPublicApi
+    public interface ModifyHttpContent extends ModifyHttpRequest {
         /**
          * The {@link SdkHttpFullRequest} that was created as a result of marshalling the {@link #request()}. This is the HTTP
          * request that will be sent to the downstream service.
          */
-        SdkHttpFullRequest httpRequest();
+        SdkHttpRequest httpRequest();
```

Comment:
Shouldn't this be in AfterMarshalling?

Reply:
See Andrew's comment on composite interceptors.